pkg/operator/statefulset.go

```diff
@@ -24,74 +24,154 @@ import (
 
 const configReloaderPort = 8080
 
+type ConfigReloader struct {
+	name           string
+	config         ReloaderConfig
+	reloadURL      url.URL
+	listenLocal    bool
+	localHost      string
+	logFormat      string
+	logLevel       string
+	additionalArgs []string
+	volumeMounts   []v1.VolumeMount
+	shard          int32
+}
+
+type option = func(*ConfigReloader)
+
+func ReloaderRunOnce() option {
+	return func(c *ConfigReloader) {
+		c.additionalArgs = append(c.additionalArgs, fmt.Sprintf("--watch-interval=%d", 0))
+	}
+}
+
+func ListenAddress() option {
+	return func(c *ConfigReloader) {
+		var listenAddress string
+		if c.listenLocal {
+			listenAddress = c.localHost
+		}
+		c.additionalArgs = append(c.additionalArgs, fmt.Sprintf("--listen-address=%s:%d", listenAddress, configReloaderPort))
+	}
+}
+
+func ReloadURL() option {
+	return func(c *ConfigReloader) {
+		c.additionalArgs = append(c.additionalArgs, fmt.Sprintf("--reload-url=%s", c.reloadURL.String()))
+	}
+}
+
+func ReloaderResources(rc ReloaderConfig) option {
+	return func(c *ConfigReloader) {
+		c.config = rc
+	}
+}
+
+func ReloaderURL(reloadURL url.URL) option {
+	return func(c *ConfigReloader) {
+		c.reloadURL = reloadURL
+	}
+}
+
+func ListenLocal(listenLocal bool) option {
+	return func(c *ConfigReloader) {
+		c.listenLocal = listenLocal
+	}
+}
+
+func LocalHost(localHost string) option {
+	return func(c *ConfigReloader) {
+		c.localHost = localHost
+	}
+}
+
+func LogFormat(logFormat string) option {
+	return func(c *ConfigReloader) {
+		c.logFormat = logFormat
+	}
+}
+
+func LogLevel(logLevel string) option {
+	return func(c *ConfigReloader) {
+		c.logLevel = logLevel
+	}
+}
+
+func AdditionalArgs(args []string) option {
```

Comment:
could we get rid of `AdditionalArgs()` and add functions for things that aren't currently represented as option?

Reply:
Thanks for the review and great feedbacks, @simonpasquier!

`func additionalArgs` created to take of options of running config-reloader; like `configReloaderArgs` in `func makeStatefulSetSpec`.

If removed; to run config-reloader just once, a bool field named, e.g.,`runOnce` might be needed in struct `ConfigReloader` for `func ReloaderRunOnce` and used in `func CreateConfigReloader` as following
```
if c.runOnce {
    args  = append(args, fmt.Sprintf("--watch-interval=%d", 0)
}
```
Addition fields in string would be needed for  `--watched-dir`, `--config-file` and etc in struct ConfigReloader. 

Feel free to correct me if misunderstand the comment and point me the enlightening path to address the comment.

Reply:
@simonpasquier Would you please give another review on this PR?